k8s/role.go

```diff
@@ -0,0 +1,67 @@
+package k8s
+
+import (
+	"time"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+)
+
+func UpdateRoleBindingFromYaml(k8sClient *kubernetes.Clientset, roleBindingYaml string) error {
+	roleBinding := rbacv1.RoleBinding{}
+	err := decodeYamlResource(&roleBinding, roleBindingYaml)
+	if err != nil {
+		return err
+	}
+
+	for retries := 0; retries <= 5; retries++ {
```

Comment:
This retry block is being repeated in several places. Consider moving it to its own separate method, and passing func() as an argument